gateway/actix_auth/src/providers/mock.rs

```diff
@@ -1,28 +1,21 @@
 use actix_web::HttpRequest;
 use serde::{Deserialize, Serialize};
 
-use super::{IdentityProvider, ProviderIdentityStatus, ProviderSessionStatus, SessionProvider};
+use super::{ProviderSessionStatus, SessionProvider};
 
 #[derive(Clone)]
 pub struct MockProvider {
-    require_login: bool,
     username: String,
     user_id: Option<String>,
 }
 
 impl MockProvider {
-    pub fn new(require_login: bool, username: String, user_id: Option<String>) -> Self {
-        Self {
-            require_login,
-            username,
-            user_id,
-        }
+    pub fn new(username: String, user_id: Option<String>) -> Self {
+        Self { username, user_id }
     }
 
     fn get_user_id(&self) -> &str {
-        self.user_id
-            .as_deref()
-            .unwrap_or_else(|| self.username.as_ref())
+        self.user_id.as_deref().unwrap_or("mocked")
```

Comment:
I think `MockProvider::user_id` should'nt be optional, and that the config parser should handle setting the default